src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/AlignmentInterval.java

```diff
@@ -5,43 +5,158 @@
 
     public final Cigar cigarAlong5to3DirectionOfContig;
 
     public final boolean forwardStrand;
+
+    /**
+     * Mapping quality, {@link SAMRecord#NO_MAPPING_QUALITY} if unspecified.
+     */
     public final int mapQual;
+
+    /**
+     * Number of mismatches, {@link #NO_NM} if unknown or unspecified.
+     */
     public final int mismatches;
+
+    /**
+     * Alignment score, {@link #NO_AS} if unknown or unspecified.
+     */
     public final int alnScore;
 
     // if any of the following boolean fields are true, fields "mapQual", "mismatches", "alnScore" should be viewed
     // with care as they were simply copied from the original alignment (not for "mismatches"),
     // which after the split are wrong (we didn't recompute them because that would require expensive SW re-alignment)
+
     public final boolean isFromSplitGapAlignment;
     public final boolean hasUndergoneOverlapRemoval;
 
+    /**
+     * Compose an alignment interval instance from a SAM supplementary alignment formatted string.
+     * <p>
+     *     The input string format is:
+     *     <pre>
+     *         chr-name,start,strand,cigar,mq,nm,as
+     *     </pre>
+     *     where mq (mapping-quality), nm (number of mismatches) and as (alignment score) might be absent. The strand
+     *     is symbolized as '+' for the forward-strand and '-' for the reverse-strand.
+     *     <p>Examples:</p>
+     *     <pre>
+     *         chr10,1241241,+,10S1313M45I14M100H,30,5,66
+     *         chr10,1241241,+,10S1313M45I14M100H,30,5
+     *         chr10,1241241,+,10S1313M45I14M100H,30
+     *         chr10,1241241,+,10S1313M45I14M100H
+     *     </pre>
+     * </p>
+     * @param str the input string.
+     * @throws IllegalArgumentException if {@code str} is {@code null} or it does not look like a valid
+     *   SA string.
+     */
+    public AlignmentInterval(final String str) {
+        Utils.nonNull(str, "input str cannot be null");
+        final String[] parts = str.replaceAll(";$", "").split(",");
+        if (parts.length < 4) {
+            throw new IllegalArgumentException("the input SA string at least must contain 4 parts");
+        }
+
+        int nextPartsIndex = 0; // holds the next element index in parts to be parsed.
+        // Coordinates:
+        final String referenceContig = parts[nextPartsIndex++];
+        final int start = Integer.parseInt(parts[nextPartsIndex++]);
+        // Strand:
+        final Strand strand = parseStrand(parts[nextPartsIndex++]);
+        final boolean forwardStrand = strand == Strand.POSITIVE;
+        // Cigar:
+        final Cigar originalCigar = parseCigar(parts[nextPartsIndex++]);
+        final Cigar cigar = forwardStrand ? originalCigar : CigarUtils.invertCigar(originalCigar);
+        // Optional fields:
+        final int mappingQuality = parts.length > nextPartsIndex ? Integer.parseInt(parts[nextPartsIndex++]) : SAMRecord.NO_MAPPING_QUALITY;
```

Comment:
I guess should capture the NFE into a IAE.

Reply:
As it is now if the part[i] does not contain a valid integer this with choke with a  NumberFormatExeception when it would be preferable fail with a IllegalArgumentException.